Jenkinsfile

```diff
@@ -14,276 +14,50 @@
-    if (line.startsWith('ARG IQ_SERVER_VERSION=')) {
-      return getShortVersion(line.substring(22))
-    }
+import com.sonatype.jenkins.shared.Expectation
+
+dockerizedBuildPipeline(
+  prepare: {
+    githubStatusUpdate('pending')
+  },
+  lint: {
+    hadolint(['./Dockerfile'])
+  },
+  setVersion: {
+    env['VERSION'] = "${env.BRANCH_NAME}-${env.BUILD_NUMBER}"
+  },
+  buildAndTest: {
+    validateExpectations([
+        new Expectation('nexus-group', 'grep', '^nexus: /etc/group', 'nexus:x:1000:'),
+        new Expectation('nexus-user', 'grep', '^nexus: /etc/passwd', 'nexus:x:1000:1000:Nexus IQ user:/opt/sonatype/nexus-iq-server:/bin/false'),
+        new Expectation('iq-process', 'ps', '-e -o command,user | grep -q ^/usr/bin/java.*nexus$ | echo $?', '0'),
+        new Expectation('application-port', 'curl', '-s --fail --connect-timeout 120 http://localhost:8070/ | echo $?', '0'),
+        new Expectation('admin-port', 'curl', '-s --fail --connect-timeout 120 http://localhost:8070/ | echo $?', '0'),
+        new Expectation('log-directory', 'ls', '-la /var/log | awk \'\$9 !~ /^\\.*$/{print \$1,\$3,\$4,\$9}\'', 'drwxr-xr-x nexus nexus nexus-iq-server'),
+        new Expectation('clm-server-log', 'test', '-f /var/log/nexus-iq-server/clm-server-log.log | echo $?', '0'),
+        new Expectation('audit-log', 'test', '-f /var/log/nexus-iq-server/audit.log | echo $?', '0'),
+        new Expectation('request-log', 'test', '-f /var/log/nexus-iq-server/request.log | echo $?', '0'),
+        new Expectation('stderr-log', 'test', '-f /var/log/nexus-iq-server/stderr.log | echo $?', '0'),
+        new Expectation('home-directory', 'ls', '-la /opt/sonatype | grep nexus-iq-server | awk \'\$9 !~ /^\\.*$/{print \$1,\$3,\$4,\$9}\'', 'drwxr-xr-x nexus nexus nexus-iq-server'),
+        new Expectation('start-script', 'test', '-f /opt/sonatype/nexus-iq-server/start.sh | echo $?', '0'),
+        new Expectation('start-script-has-java-opts', 'grep', '\'JAVA_OPTS\' /opt/sonatype/nexus-iq-server/start.sh | echo $?', '0'), 
+        new Expectation('work-directory', 'ls', '-la / | grep sonatype-work | awk \'\$9 !~ /^\\.*$/{print \$1,\$3,\$4,\$9}\'', 'drwxr-xr-x nexus nexus sonatype-work'), 
+        new Expectation('data-directory', 'test', '-d /sonatype-work/data | echo $?', '0'), 
+        new Expectation('config-directory', 'ls', '-la /etc | grep nexus-iq-server | awk \'\$9 !~ /^\\.*$/{print \$1,\$3,\$4,\$9}\'', 'drwxr-xr-x nexus nexus nexus-iq-server'), 
+        new Expectation('config-file', 'test', '-f /etc/nexus-iq-server | echo $?', '0') 
+    ])
+  },
+  testResults: ['**/validate-expectations-results.xml'],
+  vulnerabilityScan: {
+    nexusPolicyEvaluation(
+      iqApplication: 'docker-nexus-iq-server',
+      iqScanPatterns: [[scanPattern: "container:${env.DOCKER_IMAGE_ID}"]],
+      iqStage: 'develop')
+  },
+  onSuccess: {
+    githubStatusUpdate('success')
+  },
+  onFailure: {
+    githubStatusUpdate('failure')
   }
```

Comment:
For the main snapshot build, rather then updating the github status , I think it would be better to send a notification (for both build success/failure) to the iq-builds slack channel. See this example for the insight-brain [build](https://sonatype.slack.com/archives/CCA312MCZ/p1684824996982179). We can even set up a daily snapshot build, so developers are more aware of the build status (instead of just before release time).

Reply:
I would agree that adding a slack message is a good idea, `buildNotifications` is in the `iq-pipeline-library` and sends email and slack messages.  See my other note on just sending a slack message.  I do think you should keep the GitHub status updates unless you are using the GitHub branch source in your job config which also sends them.